### PR TITLE
REL-942846: Publish Relativity Server DataTransfer SDK

### PR DESCRIPTION
**Pull request by ravikonnur**
- [REL-942846](https://jira.kcura.com/browse/REL-942846): Publish Relativity Server DataTransfer SDK
- Prepared branch for the next official Relativity Server 2024 release.
-  Updated Psakefile and nuspec as per SOP
- Test Evidence: https://einstein.kcura.com/display/SDE/REL-942846-Phase+3+-+Relativity.Server.DataTransfer.SDK

### CODE REVIEW

**Review by briankesecker-relativity — COMMENTED** [review]
Change log is out-of-order chronologically.  Expected?

**Comment by ravikonnur** [comment]
> Change log is out-of-order chronologically. Expected?

We have taken change log based on the commit dates. Please suggest me if I am missing anything here.

**Review by shivarajkamadod — DISMISSED** [review]

**Review by sameerrelone — DISMISSED** [review]

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-24000.0.5
```
Comment (dhanushsathukumati) [inline version.txt:1]: Could you please check the below link to make sure the version matches the required 5000.0.0 as per versioning strategy.

https://github.com/relativityone/server-adr/blob/main/00013-server-versioning-strategy.md

File: version.txt [hunk 2]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+24000.0.6
```
Comment (dhanushsathukumati) [inline version.txt:1]: Could you please check the below link to make sure the version matches the required 5000.0.0 as per versioning strategy.

https://github.com/relativityone/server-adr/blob/main/00013-server-versioning-strategy.md

Comment (ravikonnur) [reply version.txt:1]: For Highly coupled need to consider version 24000

File: rap.yml [hunk 3]
```diff
@@ -18,8 +18,9 @@ resources:
 stages:
 - template: rapci-server.yml@azdo_templates  # Template reference
   parameters:
-    RelativityBranch: ''
+    RelativityBranch: 'release-12.3-2023-patch2'
```
Comment (briankesecker-relativity) [inline rap.yml:21]: I expect that these values do not appear in `main` branch.

File: rap.yml [hunk 4]
```diff
@@ -18,8 +18,9 @@ resources:
 stages:
 - template: rapci-server.yml@azdo_templates  # Template reference
   parameters:
-    RelativityBranch: ''
+    RelativityBranch: 'release-12.3-2023-patch2'
     TemplateName: 'aio-server-2023-ga'
+    DeployHotfixOrPatch: true
```
Comment (briankesecker-relativity) [inline rap.yml:23]: I expect that these values do not appear in `main` branch.

Comment (ravikonnur) [reply rap.yml:23]: This change came from release 2023 branch, let me know this change is needed or not in main branch

File: rap.yml (hunk 3 above)
Comment (ravikonnur) [reply rap.yml:21]: This change came from release 2023 branch, let me know this change is needed or not in main branch

Comment (briankesecker-relativity) [reply rap.yml:21]: Most server repositories keep this field as empty string.  Check this query https://github.com/search?q=org%3Arelativityone+RelativityBranch+path%3A*.yml&type=code&p=4

@scottmcburney Will this be a problem once merged to main branch?

File: rap.yml (hunk 4 above)
Comment (briankesecker-relativity) [reply rap.yml:23]: I don't find any Server repos that have this parameter defined.  https://github.com/search?q=org%3Arelativityone+DeployHotfixOrPatch+path%3A*.yml&type=code

@scottmcburney Will this be a problem once merged to main branch?

Comment (ravikonnur) [reply rap.yml:23]: @scottmcburney Any inputs on the above changes.

Comment (scottmcburney) [reply rap.yml:23]: "DeployHotfixOrPatch" was added by server-deploy.  You do NOT want it to be "True" in the main branch. 
server-deploy could supply more details.

Comment (ravikonnur) [reply rap.yml:23]: @jaganganta, Can you please share inputs on this.
Cc: @shivarajkamadod

Comment (chinmoyeep) [reply rap.yml:23]: @ravikonnur: "**DeployHotfixOrPatch**" parameter should be added only for Patch/Hotfix branch. It is documented in this Einstein page [https://einstein.kcura.com/x/c6nZJQ](https://einstein.kcura.com/x/c6nZJQ)

File: rap.yml (hunk 3 above)
Comment (ravikonnur) [reply rap.yml]: Updated RelativityBranch value to empty

File: rap.yml (hunk 4 above)
Comment (ravikonnur) [reply rap.yml]: Removed "DeployHotfixOrPatch" parameter as its not required for main branch.

**Review by shivarajkamadod — APPROVED** [review]

**Review by dhanushsathukumati — APPROVED** [review]

File: version.txt [hunk 5]
```diff
@@ -1 +1 @@
-24000.0.6
\ No newline at end of file
+24000.0.7
```
Comment (shivarajkamadod) [inline version.txt:1]: This SDK internal dependency is highly coupled with relativity.

**Review by shivarajkamadod — APPROVED** [review]

**Review by griffin-metevia — CHANGES_REQUESTED** [review]
This pull request shouldn't be merged until the release version of Relativity.Server.DataTransfer.SDK has been published.

**Comment by shivarajkamadod** [comment]
> This pull request shouldn't be merged until the release version of Relativity.Server.DataTransfer.SDK has been published.

@griffin-metevia I think that you wanted to add a comment on this PR - https://github.com/relativityone/server-datatransfer-transfer-api-legacy/pull/6.

**Review by griffin-metevia — APPROVED** [review]

**Review by scott-parillo — APPROVED** [review]

**Pull request merged**